### PR TITLE
improve the docs for V5 launch

### PR DESCRIPTION
**Pull request by ndelangen**
Issue: docs are lagging behind implementation

## What I did
Bringing the docs up to date

### CODE REVIEW

**Comment by shilman** [comment]
@ndelangen We should add redirects for the old URLs

**Review by shilman — APPROVED** [review]
Changes look great!

**Review by shilman — CHANGES_REQUESTED** [review]
We should add redirects for the old URLs

**Comment by ndelangen** [comment]
@shilman Do you have a strategy to set the redirects?

**Comment by shilman** [comment]
@ndelangen I'm using Netlify's redirects for server-side redirects: https://github.com/storybooks/frontpage/blob/master/static/_redirects

For client-side I tried something but don't think it worked: https://github.com/storybooks/frontpage/blob/master/gatsby-node.js

File: docs/package.json [hunk 1]
```diff
@@ -28,7 +28,7 @@
     "common-tags": "^1.8.0",
     "gatsby": "^1.9.279",
     "gatsby-link": "^1.6.45",
-    "gatsby-plugin-sharp": "^2.0.12",
+    "gatsby-plugin-sharp": "^1.6.48",
```
Comment (shilman) [inline docs/package.json]: @ndelangen why downgrade?

File: docs/src/pages/configurations/custom-webpack-config/index.md [hunk 2]
```diff
@@ -5,9 +5,171 @@ title: 'Custom Webpack Config'
...
 
-The file can export an object or a function. The former configures in [extend mode](#extend-mode), the latter in [full control mode](#full-control-mode), both described here.
+Storybook has its own Webpack setup and a dev server. 
+The webpack config [is configurable](/configurations/webpack), and the default can depend on which framework you're using and whether you've used a generator like [Create React App](https://github.com/facebookincubator/create-react-app) or Angular CLI etc.
 
-## Extend Mode
+> We're trying to make storybook more zero-config over time, **help to hook into the config of generators is very welcome**.
+
+<details>
+  <summary>This is what the config for storybook looks like when using CRA in dev-mode:</summary>
```
Comment (shilman) [inline docs/src/pages/configurations/custom-webpack-config/index.md]: Why do we put this in our docs? Can we just link to a file that won't get out of date every time somebody makes an change?

File: docs/src/pages/configurations/default-config/index.md [hunk 3]
```diff
@@ -38,7 +24,146 @@ So, you could use any babel plugins or presets that you have used in your projec
...
+The webpack config [is configurable](/configurations/webpack), and the default can depend on which framework you're using and whether you've used a generator like [Create React App](https://github.com/facebookincubator/create-react-app) or Angular CLI etc.
+
+> We're trying to make storybook more zero-config over time, **help to hook into the config of generators is very welcome**.
+
+<details>
+  <summary>This is what the config for storybook looks like when using CRA in dev-mode:</summary>
+
+  ```js
+  {
+    mode: 'development',
```
Comment (shilman) [inline docs/src/pages/configurations/default-config/index.md]: Duplicated??

File: docs/src/pages/configurations/theming/index.md [hunk 4]
```diff
@@ -46,57 +46,58 @@ addParameters({
...
 Next paste the code below and tweak the variables.
 
-```
+```ts
 import { create } from '@storybook/theming';
 
 export default create({
-  // Is this a 'light' or 'dark' theme?
-  base: 'light',
+  base: 'light';
```
Comment (shilman) [inline docs/src/pages/configurations/theming/index.md]: Semicolons???

File: docs/src/pages/configurations/theming/index.md [hunk 5]
```diff
@@ -105,13 +106,27 @@ addParameters({
...
 
+Many them variables are optional, the `base` property is NOT. This is a perfectly valid theme:
+
+```ts
+import { create } from '@storybook/theming';
+
+export default create({
+  base: 'light';
+
+  brandTitle: 'My custom storybook';
```
Comment (shilman) [inline docs/src/pages/configurations/theming/index.md]: semi-colons???

**Review by shilman — CHANGES_REQUESTED** [review]
See comments above

**Comment by shilman** [comment]
@ndelangen Outside the scope of this PR but we should publish the `frontpage` components as a standalone library and share them between both projects

**Comment by ndelangen** [comment]
> @ndelangen Outside the scope of this PR but we should publish the frontpage components as a standalone library and share them between both projects

I agree with the out of scope part

I'm not sure publishing these components as a library is worth our time. I mean maybe.

I think the docs should be more pure about 'content' and not about styling.
This PR is mainly to get it somewhat in sync visually, until we've integrated docs better into the build of the frontpage repo I think.

File: docs/src/new-components/layout/Header.js [hunk 6]
```diff
@@ -0,0 +1,185 @@
...
+  );
+
+  return (
+    <NavWrapper {...props}>
+      <GlobalStyle />
+
+      <Nav>
+        <NavGroup>
+          <NavItem>
+            <LogotypeWrapper isGatsby to="/">
```
Comment (shilman) [inline docs/src/new-components/layout/Header.js]: This isn't working, at least in dev mode

Comment (ndelangen) [reply docs/src/new-components/layout/Header.js]: I'm on it

File: docs/src/pages/configurations/default-config/index.md (hunk 3 above)
Comment (ndelangen) [reply docs/src/pages/configurations/default-config/index.md]: yes, didn't know a good way of having a SSOT for this.

File: docs/src/pages/configurations/custom-webpack-config/index.md (hunk 2 above)
Comment (ndelangen) [reply docs/src/pages/configurations/custom-webpack-config/index.md]: There is no single file that has this info, it's all merged from multiple presets.

I think it's valuable for people who want to customize to quickly see what we're shipping with to most users.

File: docs/package.json (hunk 1 above)
Comment (ndelangen) [reply docs/package.json]: I had some issues, I think the gatsby-plugin-sharp v2 is for gatsby v2

**Comment by ndelangen** [comment]
@shilman Good to merge?

**Comment by ndelangen** [comment]
@shilman can we please merge this, do you think?

**Review by shilman — APPROVED** [review]
LGTM! Great job 🚀

**Comment by shilman** [comment]
Adding redirects as documentation before I merge, although finally they will be applied to the frontpage directory not here

**Pull request merged**